autometa/taxonomy/ncbi.py

```diff
@@ -601,9 +629,29 @@ def search_prot_accessions(
         logger.debug(f"sseqids converted from {filename}: {converted_sseqid_count:,}")
         return sseqids_to_taxids
 
+    def search_genome_accessions(self, accessions: set):
+        sseqids_to_taxids = {}
+        fpath = self.genome_accession2taxid_fpath
+        # "rt" open the database in text mode instead of binary to be handled like a text file
+        fh = gzip.open(fpath, "rt") if fpath.endswith(".gz") else open(fpath)
+        # skip the header line
+        __ = fh.readline()
+        n_lines = file_length(fpath, approximate=True)
+        converted_sseqid_count = 0
+        logger.debug("parsing genome_accession2taxid_fpath...")
+        for line in fh:
+            acc_num, acc_ver, taxid = line.strip().split("\t")
+            taxid = int(taxid)
+            if acc_ver in accessions:
+                sseqids_to_taxids[acc_ver] = taxid
+                converted_sseqid_count += 1
+        fh.close()
+        logger.debug(f"sseqids converted: {converted_sseqid_count:,}")
+        return sseqids_to_taxids
```

Comment:
This appears to only be relevant to GTDB. Why not create a `gtdb.py` file and subclass NCBI if you would like to use the NCBI class utilities? Handling for both NCBI and GTDB in the same file is going to be difficult to continue to extend if there are other databases in the future that wish to be integrated. It may be simpler, easier to understand and perhaps more effective to keep each of the respective handling of taxonomy databases to their own corresponding files.